### PR TITLE
Do not require rekor and fulcio data

### PR DESCRIPTION
**Pull request by flavio**
Make rekor and fulcio params not mandatory when creating a new instance of the `cosign::Client`.

This fixes issue #17

### CODE REVIEW

File: src/cosign/mod.rs [hunk 1]
```diff
@@ -182,17 +207,28 @@ impl ClientBuilder {
...
-        let rekor_pub_key = crypto::new_verification_key(&rekor_pub_key_raw)?;
+        let rekor_pub_key = match self.rekor_pub_key {
+            None => {
+                info!("rekor public key not provided");
+                None
+            }
+            Some(raw) => Some(crypto::new_verification_key(&raw)?),
+        };
+
+        let fulcio_pub_key_raw = match self.fulcio_cert {
```
Comment (viccuad) [inline src/cosign/mod.rs:225]: Nit:

```suggestion
        let fulcio_pub_key_der = match self.fulcio_cert {
```

File: src/cosign/mod.rs [hunk 2]
```diff
@@ -257,8 +286,14 @@ impl CosignCapabilities for Client {
     ) -> Result<Vec<SimpleSigning>> {
         let (manifest, layers) = self.fetch_manifest_and_layers(auth, cosign_image).await?;
 
-        let (_, fulcio_pub_key) = SubjectPublicKeyInfo::from_der(&self.fulcio_pub_key_raw)
-            .map_err(|e| anyhow!("Cannot parse fulcio public key: {:?}", e))?;
+        let fulcio_pub_key = match &self.fulcio_pub_key_raw {
+            None => None,
+            Some(raw) => {
```
Comment (viccuad) [inline src/cosign/mod.rs:291]: Nit:

```suggestion
            Some(der) => {
```

File: src/cosign/signature_layers.rs [hunk 3]
```diff
@@ -217,8 +229,8 @@ impl SignatureLayer {
 pub(crate) fn build_signature_layers(
```
Comment (viccuad) [inline src/cosign/signature_layers.rs:229]: The rustdoc comment of this function may not be true anymore, as the rekor and fulcio keys may be `None`. In that case, when we call `SignatureLayer::new()` on building the `signature_layers`, we are creating layers that only have the `signature` and no `bundle` or `certificate_key`, which is ok, as we can go and verify the signature. But maybe is worth noting in the rustdoc.

**Review by viccuad — APPROVED** [review]
LGTM modulo silly nitpicks. The untouched code has the correct precedence and outputs so verification should work as expected when the rekor and fulcio data is None, and we have tests for it.

Note, I'm still learning about the whole workflow, and just learnt about cosign bundles.

**Review by lukehinds — APPROVED** [review]

File: src/cosign/signature_layers.rs (hunk 3 above)
Comment (flavio) [reply src/cosign/signature_layers.rs:229]: Good catch, I fixed that now

**Comment by flavio** [comment]
Thanks for all the reviews

**Pull request merged**